server/internal/middleware/chatSession.go

```diff
@@ -0,0 +1,37 @@
+package middleware
+
+import (
+	"net/http"
+	"slices"
+
+	"github.com/speakeasy-api/gram/server/internal/auth/chatsessions"
+	"github.com/speakeasy-api/gram/server/internal/auth/constants"
+	"github.com/speakeasy-api/gram/server/internal/contextvalues"
+)
+
+// ChatSessionMiddleware validates the chat session token and the request origin is in the allowed origins
+// TODO: only allow this on certain routes?
```

Comment:
That's what I was hoping we'd achieve by isolating it as a separate middleware. I expected it to be used in the chat service's `Attach` function.

Reply:
Ah I see 🤔